import_export/resources.py

```diff
@@ -1292,13 +1293,20 @@ class ArrayField:
         return result
 
     @classmethod
-    def widget_kwargs_for_field(self, field_name):
+    def widget_kwargs_for_field(cls, field_name, django_field):
         """
         Returns widget kwargs for given field_name.
         """
-        if self._meta.widgets:
-            return self._meta.widgets.get(field_name, {})
-        return {}
+        widget_kwargs = {}
+        if cls._meta.widgets:
+            cls_kwargs = cls._meta.widgets.get(field_name, {})
+            widget_kwargs.update(cls_kwargs)
+        if (
+            issubclass(django_field.__class__, fields.CharField)
+            and django_field.blank is True
+        ):
+            widget_kwargs.update({"coerce_to_string": True, "allow_blank": True})
```

Comment:
Hi Ryan, thanks for reviewing, much appreciated.  

> How will this behave if null and blank are both true?

'blank' is the overriding setting - so it will store `None` as empty string.

> Basically, if the field allows blanks, we treat all nulls as blanks on export, and empty fields/cells as blanks on import. 

Yes, this is the nub of it.  This change only affects imports.  There are other changes planned to make exports export either string or raw type.

Users can override the behaviour added in this PR by declaring their own Field, e.g. this declaration in `core/admin.py` will cause the original error to re-occur.

```
    author_email = fields.Field(
        attribute="author_email",
        widget=widgets.CharWidget(allow_blank=False, coerce_to_string=False))
```